packages/experimental/Link/src/useLink.ts

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { useAsPressable, useKeyProps, useOnPressWithFocus, useViewCommandFocus } from '@fluentui-react-native/interactive-hooks';
+import { LinkProps, LinkState } from './Link.types';
+import { memoize } from '@fluentui-react-native/framework';
+import { AccessibilityState, Linking, Platform } from 'react-native';
+
+export const useLink = (props: LinkProps): LinkState => {
+  // attach the pressable state handlers
+  const defaultComponentRef = React.useRef(null);
+  const {
+    accessibilityRole,
+    onPress,
+    onAccessibilityTap,
+    tooltip,
+    url,
+    accessibilityState,
+    componentRef = defaultComponentRef,
+    disabled,
+    enableFocusRing,
+    focusable,
+    ...rest
+  } = props;
+  const isDisabled = !!disabled;
+
+  const [visitedState, setVisitedState] = React.useState({ visited: false });
+  const linkOnPress = React.useCallback(
+    (e) => {
+      setVisitedState({ visited: true });
+      if (url) {
+        Linking.openURL(url as string);
+      } else if (onPress) {
+        onPress(e);
+      }
+      e.stopPropagation();
+    },
+    [setVisitedState, url, onPress],
+  );
+
+  // GH #1336: Set focusRef to null if link is disabled to prevent getting keyboard focus.
+  const focusRef = isDisabled && !focusable ? null : componentRef;
+  const onPressWithFocus = useOnPressWithFocus(focusRef, linkOnPress);
+  const pressable = useAsPressable({ ...rest, disabled: isDisabled, onPress: onPressWithFocus });
```

Comment:
Saad's been working on switching things to use Pressable/usePressableState. Is that possible with Link or not really because of the Text aspect?

Reply:
As an example: https://github.com/microsoft/fluentui-react-native/pull/2156/files#diff-209b8ee779e86a23bc406c0d0dff9ab5275c85ff6a9197095fafdc89e6ea24a3

Reply:
Can I make this change in a separate PR along with the pressable. I might have to change how Link handles pressable  events with the changes in Text.

Reply:
[https://office.visualstudio.com/OC/_sprints/taskboard/CXE%20React/OC/CY22Q3%20(Jul-Sep)/2208-Aug/Aug02?workitem=6894889](work item for switching Text and Link to Pressable)

Reply:
Yeah for sure

Reply:
@saadnajmi FYI